sound/soc/sof/sof-audio.c

```diff
@@ -11,6 +11,66 @@
 #include "sof-audio.h"
 #include "ops.h"
 
+static int sof_kcontrol_setup(struct snd_sof_dev *sdev, struct snd_sof_control *scontrol)
+{
+	int ipc_cmd, ctrl_type;
+	int ret;
+
+	/* reset readback offset for scontrol */
+	scontrol->readback_offset = 0;
+
+	/* notify DSP of kcontrol values */
+	switch (scontrol->cmd) {
+	case SOF_CTRL_CMD_VOLUME:
```

Comment:
do we need the "fallthrough"'s here?

Reply:
no, we only need fallthrough if its not straightforward, for example when we do something for VOLUME but then continue on to do more common stuff for volume/switch/enum.